src/Data/ProgrammesDb/EntityRepository/ImageRepository.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class ImageRepository extends EntityRepository
+{
+    public function findByPid(string $pid): ?array
+    {
+        $qb = parent::createQueryBuilder('image')
+            ->andWhere('image.pid = :pid')
+            ->andWhere('image.isEmbargoed = 0')
```

Comment:
This should be automatically added to all embargoable entities by the [EmbargoedFilter](https://github.com/bbc/programmes-pages-service/blob/master/src/Data/ProgrammesDb/Filter/EmbargoedFilter.php), so you don't need to explicitly specify it.

Reply:
Is that automatic? Or do I need to add a trait or annotation or something?

Reply:
It's automatic based on if an entity has a field called isEmbargoed yep